EndlessClient/Dialogs/BarberDialog.cs

```diff
@@ -0,0 +1,217 @@
+                                       dialogButtonService.GetSmallDialogButtonOutSource(buttonType),
+                                       dialogButtonService.GetSmallDialogButtonOverSource(buttonType));
+            button.Initialize();
+            button.SetParentControl(this);
+            return button;
+        }
+
+        private void UpdateListItems(CharacterRenderProperties currentProperties)
+        {
+            _changeHairItem.PrimaryText = _localizedStringFinder.GetString(EOResourceID.DIALOG_BARBER_CHANGE_MODAL);
+            _changeHairItem.SubText = GetCurrentHairStyleText(currentProperties.HairStyle);
+
+            _changeHairColor.PrimaryText = _localizedStringFinder.GetString(EOResourceID.DIALOG_BARBER_CHANGE_HAIR_COLOR);
+            _changeHairColor.SubText = GetCurrentHairColorText(currentProperties.HairColor);
+
+            _changeBuyHairStyleOrColor.PrimaryText = _localizedStringFinder.GetString(EOResourceID.DIALOG_BARBER_BUY_HAIRSTYLE);
+            _changeBuyHairStyleOrColor.SubText = GetCostText();
+        }
+
+        private string GetCurrentHairStyleText(int hairStyle) => $"{_localizedStringFinder.GetString(EOResourceID.DIALOG_WORD_CURRENT)}: {hairStyle}";
+        private string GetCurrentHairColorText(int hairColor) => $"{_localizedStringFinder.GetString(EOResourceID.DIALOG_WORD_CURRENT)}: {_hairColorNames[hairColor % _hairColorNames.Length]}";
+        private string GetCostText() => $"{_localizedStringFinder.GetString(EOResourceID.DIALOG_WORD_CURRENT)}: {CalculateCost()} gold";
+
+        private int CalculateCost()
+        {
+            var level = (int)_characterRepository.MainCharacter.Stats[CharacterStat.Level];
+            return 200 + Math.Max(level - 1, 0) * 200;
+        }
+
+        private void Hair(object sender, MonoGame.Extended.Input.InputListeners.MouseEventArgs e)
+        {
+            _characterControl.NextHairStyle();
+            _changeHairItem.SubText = GetCurrentHairStyleText(_characterControl.RenderProperties.HairStyle);
+        }
+
+        private void HairColor(object sender, MonoGame.Extended.Input.InputListeners.MouseEventArgs e)
+        {
+            UpdateHairColor();
+        }
+        private void UpdateHighlightWidth(ListDialogItem item)
+        {
+            item.HighlightWidthOverride = 175;
+        }
+        private void UpdateHairColor()
+        {
+            var currentProperties = _characterControl.RenderProperties;
+            _characterControl.NextHairColor();
+            _changeHairColor.SubText = GetCurrentHairColorText(_characterControl.RenderProperties.HairColor);
+        }
+
+        private void BuyHairStyleOrColor(object sender, MonoGame.Extended.Input.InputListeners.MouseEventArgs e) => PurchaseHairStyleOrColor(_characterControl.RenderProperties.HairStyle, _characterControl.RenderProperties.HairColor);
+
+        private void PurchaseHairStyleOrColor(int hairStyle, int hairColor)
```

Comment:
Also prefer to inline this method. I don't think having the event handles hook into a different method really adds much clarity.